Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/StateSynchronization/StateSynchronizationBroadcaster.cs

```diff
@@ -108,38 +103,30 @@ private void DebugLog(string message)
         {
             if (debugLogging)
             {
-                Debug.Log($"StateSynchronizationBroadcaster - {connectedObservers.Count} Connections : {message}");
+                Debug.Log($"StateSynchronizationBroadcaster: {message}");
             }
         }
 
         protected void OnConnected(SocketEndpoint endpoint)
         {
             DebugLog($"Broadcaster received connection from {endpoint.Address}.");
 
-            ConnectedObserver connectedObserver = new ConnectedObserver(Role.Broadcaster, endpoint, () => Instantiate(anchorPrefab), debugLogging);
-            connectedObservers.Add(endpoint, connectedObserver);
             Connected?.Invoke(endpoint);
 
-            DebugLog($"Broadcaster kicking off spatial localization on observer from {endpoint.Address}.");
-
-            if (SpatialLocalizationMechanism != null)
+            foreach (var handler in allHandlers)
```

Comment:
>foreach (var handler in allHandlers) [](start = 12, length = 36)

Right now these command-raising loops are not reentrancy safe. If anyone tries to call Register or Unregister while in an OnConnected/OnDisconnected/HandleCommand, it will modify the collection during enumeration.

You could make this safe by either iterating over a copy of the list each time, or by putting a guard against reentrancy so that calls to Register/Unregister perhaps put the new connections into a "pending changes" list if iteration is happening.

This might be design overkill, but I've been bitten by this particular issue a lot in the past.